morango/controller.py

```diff
@@ -1,14 +1,24 @@
 import json
 
-from django.core.exceptions import ObjectDoesNotExist
 from django.core.serializers.json import DjangoJSONEncoder
 from django.db import transaction
+from django.db.models import Q
 from django.utils.six import iteritems
 from morango.models import DeletedModels, InstanceIDModel, RecordMaxCounter, Store
 
 from morango.utils.register_models import _profile_models
 
 
+def _self_referential_fk(klass_model):
+    """
+    Return whether this model has a self ref FK, and the name for the field
+    """
+    for f in klass_model._meta.concrete_fields:
+        if f.related_model == klass_model:
+            return True, f.attname
+    return False, None
```

Comment:
This works, but might be enough to just return the attname or None, since they're appropriately booleanizable themselves?

Reply:
Also, perhaps good to throw an error if there are two self-references?